src/include/gurt/debug.h

```diff
@@ -63,7 +63,7 @@ extern void (*d_alt_assert)(const int, const char*, const char*, const int);
  * scope.   This enables use of a variable in the macro below and is
  * just good coding practice.
  */
-#pragma GCC diagnostic warning "-Wshadow"
+#pragma GCC diagnostic error "-Wshadow"
```

Comment:
Wouldn't we also need to fix the existing instances? Such as this one that's failing the builds
```
src/object/cli_shard.c:1906
```